I break it into 2 cases, I think that's easier than using assume. However, I got this error:
```
======================================================================
ERROR: test_lzma_round_trip_format_xz (tests.test_compression.TestLZMA)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/cheuktingho/Contribution/stdlib-property-tests/tests/test_compression.py", line 53, in test_lzma_round_trip_format_xz
    payload=st.binary(),
  File "/Users/cheuktingho/.pyenv/versions/stdlib-test/lib/python3.7/site-packages/hypothesis/core.py", line 1102, in wrapped_test
    raise the_error_hypothesis_found
  File "/Users/cheuktingho/Contribution/stdlib-property-tests/tests/test_compression.py", line 62, in test_lzma_round_trip_format_xz
    payload, format="FORMAT_XZ", check=check, preset=compresslevel
  File "/Users/cheuktingho/.pyenv/versions/3.7.3/lib/python3.7/lzma.py", line 318, in compress
    comp = LZMACompressor(format, check, preset, filters)
TypeError: an integer is required (got type str)
```
Which I don't understand when I look at the source code: https://github.com/python/cpython/blob/c1f1ddf30a595c2bfa3c06e54fb03fa212cd28b5/Lib/zipfile.py#L612

Can't see where the arguments are taken here.

--- a/tests/test_compression.py
+++ b/tests/test_compression.py
@@ -77,9 +77,9 @@ class TestLZMA(unittest.TestCase):
     @given(
         payload=st.binary(),
         data=st.data())
-    def test_lzma_round_trip_format_alone(self, payload, data):
+    def test_lzma_round_trip_format_raw(self, payload, data):
         # create the list of filter ids
-        filter_ids=data.draw(st.lists([lzma.FILTER_DELTA, lzma.FILTER_X86, lzma.FILTER_IA64, lzma.FILTER_ARM, lzma.FILTER_ARMTHUMB, lzma.FILTER_POWERPC, lzma.FILTER_SPARC], max_size=3))
+        filter_ids=data.draw(st.lists(st.sampled_from([lzma.FILTER_DELTA, lzma.FILTER_X86, lzma.FILTER_IA64, lzma.FILTER_ARM, lzma.FILTER_ARMTHUMB, lzma.FILTER_POWERPC, lzma.FILTER_SPARC]), max_size=3))
         filter_ids.append(lzma.FILTER_LZMA2)
         # create filters options
         filters=[]
@@ -96,7 +96,7 @@ class TestLZMA(unittest.TestCase):
                             "depth":data.draw(st.integers(min_value=0))
                             })
         result = lzma.decompress(
-            lzma.compress(payload, format=lzma.FORMAT_ALONE, filters=filters)
+            lzma.compress(payload, format=lzma.FORMAT_RAW, filters=filters)
         )
         self.assertEqual(payload, result)
 

--- a/tests/test_compression.py
+++ b/tests/test_compression.py
@@ -51,22 +51,23 @@ class TestLZMA(unittest.TestCase):
     # TODO: https://docs.python.org/3/library/lzma.html
     @given(
         payload=st.binary(),
+        format=st.just(lzma.FORMAT_XZ),
         check=st.sampled_from(
-            ["CHECK_NONE", "CHECK_CRC32", "CHECK_CRC64", "CHECK_SHA256"]
+            [lzma.CHECK_NONE, lzma.CHECK_CRC32, lzma.CHECK_CRC64, lzma.CHECK_SHA256]
         ),
         compresslevel=st.integers(0, 9),
     )
-    def test_lzma_round_trip_format_xz(self, payload, check, compresslevel):
+    def test_lzma_round_trip_format_xz(self, payload, format, check, compresslevel):
         result = lzma.decompress(
             lzma.compress(
-                payload, format="FORMAT_XZ", check=check, preset=compresslevel
+                payload, format=format, check=check, preset=compresslevel
             )
         )
         self.assertEqual(payload, result)
 
     @given(
         payload=st.binary(),
-        format=st.sampled_from(["FORMAT_ALONE", "FORMAT_RAW"]),
+        format=st.sampled_from([lzma.FORMAT_ALONE, lzma.FORMAT_RAW]),
         compresslevel=st.integers(0, 9),
     )
     def test_lzma_round_trip_format_others(self, payload, format, compresslevel):

--- a/tests/test_compression.py
+++ b/tests/test_compression.py
@@ -51,31 +51,54 @@ class TestLZMA(unittest.TestCase):
     # TODO: https://docs.python.org/3/library/lzma.html
     @given(
         payload=st.binary(),
-        format=st.just(lzma.FORMAT_XZ),
         check=st.sampled_from(
             [lzma.CHECK_NONE, lzma.CHECK_CRC32, lzma.CHECK_CRC64, lzma.CHECK_SHA256]
         ),
         compresslevel=st.integers(0, 9),
     )
-    def test_lzma_round_trip_format_xz(self, payload, format, check, compresslevel):
+    def test_lzma_round_trip_format_xz(self, payload, check, compresslevel):
         result = lzma.decompress(
             lzma.compress(
-                payload, format=format, check=check, preset=compresslevel
+                payload, format=lzma.FORMAT_XZ, check=check, preset=compresslevel
             )
         )
         self.assertEqual(payload, result)
 
     @given(
         payload=st.binary(),
-        format=st.sampled_from([lzma.FORMAT_ALONE, lzma.FORMAT_RAW]),
         compresslevel=st.integers(0, 9),
     )
-    def test_lzma_round_trip_format_others(self, payload, format, compresslevel):
+    def test_lzma_round_trip_format_alone(self, payload, format, compresslevel):
         result = lzma.decompress(
-            lzma.compress(payload, format=format, preset=compresslevel)
+            lzma.compress(payload, format=lzma.FORMAT_ALONE, preset=compresslevel)
         )
         self.assertEqual(payload, result)
 
+    @given(
+        payload=st.binary(),
+        data=st.data())
+    def test_lzma_round_trip_format_alone(self, payload, data):
+        # create the list of filter ids
+        filter_ids=data.draw(st.lists([lzma.FILTER_DELTA, lzma.FILTER_X86, lzma.FILTER_IA64, lzma.FILTER_ARM, lzma.FILTER_ARMTHUMB, lzma.FILTER_POWERPC, lzma.FILTER_SPARC], max_size=3))
+        filter_ids.append(lzma.FILTER_LZMA2)
+        # create filters options
+        filters=[]
+        for filter in filter_ids:
+            lc = data.draw(st.integers(0, 4))
+            lp = data.draw(st.integers(0, 4-lc))
+            filters.append({"id":filter,
+                            "preset":data.draw(st.integers(0, 9)),
+                            "dict_size":data.draw(st.integers(4000, 1.875e+8)),
+                            "lc":lc,
+                            "lp":lp,
+                            "mode":data.draw(st.sampled_from([lzma.MODE_FAST, lzma.MODE_NORMAL])),
+                            "mf":data.draw(st.sampled_from([lzma.MF_HC3, lzma.MF_HC4, lzma.MF_BT2, lzma.MF_BT3, lzma.MF_BT4])),
+                            "depth":data.draw(st.integers(min_value=0))
+                            })
+        result = lzma.decompress(
+            lzma.compress(payload, format=lzma.FORMAT_ALONE, filters=filters)
+        )
+        self.assertEqual(payload, result)
 
 class TestZlib(unittest.TestCase):
     # TODO: https://docs.python.org/3/library/zlib.html

